src/runtime_src/core/tools/common/JSONConfigurable.cpp

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "tools/common/JSONConfigurable.h"
+#include "core/common/error.h"
+
+// Find children corresponding to target(s) and return them in a ptree.
+boost::property_tree::ptree
+JSONConfigurable::parse_configuration(
+  const std::vector<std::string>& targets,
+  const boost::property_tree::ptree& configuration)
+{
+  boost::property_tree::ptree output;
+  try {
+    for (const std::string& target: targets) {
+      for (const auto& configTree : configuration) {
+        for (const auto& config : configTree.second.get_child("contents")) {
+          const auto& pt = config.second;
+          if (boost::iequals(target, pt.get<std::string>("name")))
+            output.push_back(std::make_pair("", pt));
+        }
+      }
+    }
+  } catch (const std::exception& e) {
+    throw std::runtime_error(e.what());
+  }
+  return output;
+}
+
+// Parse configuration for children corresponding to current device categories before parsing for target(s).
+boost::property_tree::ptree
+JSONConfigurable::parse_configuration_tree(
+  const std::vector<std::string>& targets,
+  const boost::property_tree::ptree& configuration)
+{
+  // currentDeviceCategories must be updated to use a query for device categories of cards present.
+  const std::vector<std::string> currentDeviceCategories = {"common", "alveo", "aie"};
```

Comment:
It should be a private final member variable instead of being defined here and in `exactMatchingConfigurations(...)`.

Reply:
This will be replaced by a system query later on as per the comment above. This is a place holder until that point.